CorsixTH/Lua/entities/machine.lua

```diff
@@ -93,91 +96,130 @@ local function setSmoke(self, isSmoking)
-    -- Clear dynamic info (tracks machine usage which is no longer required)
-    self:clearDynamicInfo()
-    -- Prevent the machine from smoking, it's now just a pile of rubble
-    setSmoke(self, false)
-    -- If we have the window for this machine open, close it
-    local window = self.world.ui:getWindow(UIMachine)
-    if window and window.machine == self then
-      window:close()
-    end
-    -- Clear the icon showing a handyman is coming to repair the machine
-    self:setRepairing(nil)
+  local remaining_use_count = self:getRemainingUses()
+  local repair_task_index = self.hospital:getIndexOfTask(self.tile_x, self.tile_y, "repairing")
+  local must_explode = self:calculateIsMachineMustExplode(room, remaining_use_count)
+  if must_explode then
+    -- Room failed to be saved, it must be explode
+    self:explodeMachine(room)
     return true
-  -- Else if urgent repair needed or room didn't explode
-  elseif threshold < 4 then
+  else
+    -- Else if urgent repair needed or room didn't explode
+    self:checkIsNeedToCallHandymanForRepair(room, remaining_use_count, repair_task_index)
+    -- Update whether smoke gets displayed for this machine (and if so, how much)
+    self:calculateSmoke(room)
+  end
+end
+
+--! Call on machine used. After machine used increment use values accordingly.
+function Machine:incrementUsedCount()
+  self.times_used = self.times_used + 1
+  self.total_usage = self.total_usage + 1
+end
+
+--! Call on machine use. Handles crashing the machine & queueing repairs
+--!param room (object) machine room
+--!param remaining_use_count (int) machine remaining use count
+--!param repair_task_index (int) task id of repairing this machine handyman job
+function Machine:checkIsNeedToCallHandymanForRepair(room, remaining_use_count, repair_task_index)
+  local repair_task_dont_exist = repair_task_index == -1
+  if remaining_use_count < 4 then
     -- If the job of repairing the machine isn't queued, queue it now (higher priority)
-    if taskIndex == -1 then
+    if repair_task_dont_exist then
       local call = self.world.dispatcher:callForRepair(self, true, false, true)
       self.hospital:addHandymanTask(self, "repairing", 2, self.tile_x, self.tile_y, call)
       self.hospital:announceRepair(room)
     else -- Otherwise the task is already queued. Increase the priority to above that of machines with at least 4 uses left
        -- Upgrades task from low (1) priority to high (2) priority
        -- This does not lock the room, as happens when the task call starts at high priority
-      if self.hospital:getHandymanTaskPriority(taskIndex, "repairing") == 1 then
-        self.hospital:modifyHandymanTaskPriority(taskIndex, 2, "repairing")
+      if self.hospital:getHandymanTaskPriority(repair_task_index, "repairing") == 1 then
+        self.hospital:modifyHandymanTaskPriority(repair_task_index, 2, "repairing")
         self.hospital:announceRepair(room)
       end
     end
   -- Else if repair is needed, but not urgently
```

Comment:
```suggestion
  -- Else if (low priority) repair is needed, make sure there is a handyman task for it.
```
Combine that with what the code is doing.

Reply:
> -- Else if (low priority) repair is needed, make sure there is a handyman task for it.

Reasonable.